schema_editor/app/scripts/schemas/schemas-service.js

```diff
@@ -64,7 +96,7 @@
                             },
                             enumSource: [{
                                     source: 'target',
-                                    title: fieldData.referenceTarget + ' {{i}}',
+                                    title: displayProperties.join(' '),
```

Comment:
I think that would be good, but it would probably require pulling in a more powerful templating engine than the built-in one so that we have access to some more powerful features, because in this part of the code there's no visibility into what the eventual values might be. See the Swig example here: https://github.com/jdorn/json-editor#enum-values